frontend/src/App.tsx

```diff
@@ -17,6 +17,41 @@ function App(): ReactElement {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   const global = useContext(store) as any;
   const state = global.state;
+  const dispatch = global.dispatch;
+
+  useEffect(() => {
+    //establish websocket connection
+    const websocket: WebSocket = new WebSocket(
+      `${process.env.REACT_APP_WEBSOCKET_URL}`
+    );
+
+    //when established, join the room as the professor
+    websocket.onopen = () => {
+      websocket.send(
+        JSON.stringify({
+          action: "professorJoinRoom",
+          courseId: state.courseId,
+        })
+      );
+    };
+
+    //set the global websocket
+    dispatch({ type: "set-websocket", payload: websocket });
+
+    // if the professor closes the window, remove them from the room
+    window.onbeforeunload = () => {
+      websocket.send(
+        JSON.stringify({
+          action: "leaveRoom",
+          courseId: state.courseId,
+        })
+      );
+    };
+
+    // disable-scroll for app
+    document.body.style.overflow = "hidden";
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why disable scroll for the app in js instead of CSS? When I take that out I don't get any overflow. If the app is causing overflow in some way that's something we should fix in our CSS. And why disable the exhaustive deps here instead of adding in the deps it suggested?